graphein/ml/datasets/pdb_data.py

```diff
@@ -23,6 +25,11 @@
 )
 from graphein.utils.dependencies import is_tool
 
+PRIMARY_INTERCHAIN_CONTACT_ATOMS_FOR_FILTERING: List[str] = ["CA", "C4'"]
+SECONDARY_INTERCHAIN_CONTACT_ATOMS_NOT_FOR_FILTERING: List[str] = ["H"]
+PRIMARY_HYDROGEN_BOND_ATOMS_FOR_FILTERING: List[str] = ["N", "O", "N1", "N9", "N3", "C2", "C4", "C5", "C6"]
```

Comment:
This should be vetted more carefully, as I initially chose these atom types heuristically.

Reply:
What is this atom naming scheme? It doesn't ring any bells for me (https://github.com/a-r-j/graphein/blob/281ce30cf424f10687eb3e3df73e8888547bd98c/graphein/protein/resi_atoms.py#L276)

Reply:
We already have these constants:

https://github.com/a-r-j/graphein/blob/281ce30cf424f10687eb3e3df73e8888547bd98c/graphein/protein/resi_atoms.py#L858

https://github.com/a-r-j/graphein/blob/281ce30cf424f10687eb3e3df73e8888547bd98c/graphein/protein/resi_atoms.py#L880

Reply:
> What is this atom naming scheme? It doesn't ring any bells for me (
> 
> https://github.com/a-r-j/graphein/blob/281ce30cf424f10687eb3e3df73e8888547bd98c/graphein/protein/resi_atoms.py#L276
> 
> )

The `N`, `CA`, `O`, and `H` atoms correspond to regular protein vocabulary, however, all other types correspond to nucleic acid residue atoms. My initial goal with this PR was to make a generic dataset chain filter for protein-protein interactions, protein-nucleic acid interactions, and nucleic acid-nucleic acid interactions (inspired by the dataset curation technique of RoseTTAFold2NA for protein-nucleic acid structure prediction - https://www.biorxiv.org/content/10.1101/2022.09.09.507333v1.full.pdf - page 8).

Reply:
Per a suggestion from a colleague, I have removed the C atoms from the hydrogen bond calculation, as these atoms are very rarely involved in the formation of h-bonds in proteins and NAs.

Reply:
> The N, CA, O, and H atoms correspond to regular protein vocabulary, however, all other types correspond to nucleic acid residue atoms.

Got it, bells ring for me now :) 

So these H-bond definitions do not account for sidechain-X hbonds, only backbone-backbone hbonds?

Reply:
Right. Here's a naive question on my part: How frequent would you say the occurrence of sidechain-X hbonds is?

Reply:
Seemingly quite common!

https://academic.oup.com/peds/article/13/4/227/1627008

Reply:
By way of how I have designed this filtering logic, I am assuming that each (protein or NA) residue (potentially) contains the following atoms: `"N", "O", "N1", "N9", "N3"`. Given the prevalence of sidechain hbonds, what types of protein atoms (shared across all residue types) would you say would be most reasonable to include to cover most of the possible hbonds mentioned in this article? The only other atom type I think we could include would be the carbon-beta (Cb) atoms.